protobuf/bess_msg.proto

```diff
@@ -400,18 +400,22 @@ message GetModuleInfoRequest {
 }
 
 message GetModuleInfoResponse {
+  message GateHook {
+    string class_name = 1;         /// gate hook class_name and
+    string hook_name = 2;          /// gate hook name
+  }
   message IGate {
     message OGate {
       uint64 ogate = 1;  /// Output gate of "previous" module
       string name = 2;   /// Name of "previous" module
     }
-    uint64 igate = 1;           /// Input gate ID
-    repeated OGate ogates = 2;  /// The list of upstream output gates
-    uint64 cnt = 3;             /// # of packet batches seen
-    uint64 pkts = 4;            /// # of packets seen
-    uint64 bytes = 5;            /// # of bytes seen
-    double timestamp = 6;       /// The time that cnt/pkts counters were read
-    repeated string hook_name = 7;         /// List of hooks per gate
+    uint64 igate = 1;                /// Input gate ID
+    repeated OGate ogates = 2;       /// The list of upstream output gates
+    uint64 cnt = 3;                  /// # of packet batches seen
+    uint64 pkts = 4;                 /// # of packets seen
+    uint64 bytes = 5;                /// # of bytes seen
+    double timestamp = 6;            /// The time that cnt/pkts counters were read
+    repeated GateHook gatehooks = 7;  /// List of gate hook
```

Comment:
Sorry. I didn't get why we should renumber or reserve. 😿 Could you explain in more detail, please?

Reply:
We talked about this offline but I'll just record the summary here for posterity 😸 ... in google protobufs, there's no automatic backwards compatibility, but you can reserve the numeric value that was used in an older protocol message, and hence detect client/server protocol mismatches. If you re-use the number, an old client sending to a new server (or vice versa) can misinterpret the message.

(In this case it's not very important since the interior parts of the item won't line up—the old one was a String and the new one is a sub-message—so the client and server will notice that they're not communicating successfully anyway.  Reserving would just make it clear that there was a change at some point.  We have not been good about this in the past...)